Share action card

mendelB opened the pull request:
## _PULL REQUEST OVERVIEW_
Changing up the Share Action to be more awesome and share-able!


### What does this PR do?
- Adds Card styling and formatting based on the mocks
- Adds single link format  instead of the old multi-link format, and optional `content` field


### Any background context you want to provide?
Most of the important stuff is already in place for this! So this is just adding the card-based formatting and styling, and changing up from the original multi share link format to be just one `link`, `title`, `content` (Not completely sold on the naming of `content`. Should that be `description`? or `additionalInfo` or something?).

~Right now the Social Share card would be added like so:~
~[Example Share Action in Contentful](https://app.contentful.com/spaces/81iqaqpfd8fy/entries/2pIlJmKboMW0U2IGgM8UYu)~
~*Additional Content:*~
```
{
	"link": "https://yahoo.com",
	"title": "Share this link pretty please!"
       	"content": "*Share this link because things, and also fun stuff.*"
}
```
^^NO!
We've added contentful fields for `title` and `link` as well as the optional `content` markdown field.

Which would yield:
Large:
![image](https://user-images.githubusercontent.com/12417657/35821497-22d49dfa-0a77-11e8-920d-52cb5f81b48d.png)


Medium:
![image](https://user-images.githubusercontent.com/12417657/35821508-2fd44780-0a77-11e8-8a74-9e2b6bc89160.png)


Small:
![image](https://user-images.githubusercontent.com/12417657/35821518-38705c6c-0a77-11e8-983c-f7c2e07f6c0d.png)


Without the `content` it'd render just the same minus the content bit at the top.

### Question:
Should we go ahead and formalize these fields in Contentful / Entry in Phoenix?


### What are the relevant tickets/cards?
[#154525994](https://www.pivotaltracker.com/story/show/154525994)
[multi link --> single link Slack convo](https://dosomething.slack.com/archives/C2BPA7M8F/p1517846439000229)
[Design](https://s3.amazonaws.com/prod.tracker2/resource/86918413/Social%20Shares.png?response-content-disposition=inline&response-content-type=image%2Fpng&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAJJBSFJ4TCVKKGAIA%2F20180205%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180205T152515Z&X-Amz-Expires=1800&X-Amz-SignedHeaders=host&X-Amz-Signature=adfc54c6ca88941f180571d00626e7da348a751d663e8648a2b877c92e0d72b9)


### Checklist
- [x] Added screenshot to PR description of related front-end updates on **small** screens.
- [x] Added screenshot to PR description of related front-end updates on **medium** screens.
- [x] Added screenshot to PR description of related front-end updates on **large** screens.

mendelB commented on .env.example:
```diff
@@ -64,3 +64,5 @@ GOOGLE_ANALYTICS_ID=
 CUSTOMER_IO_ID=
 PUCK_URL=
+
+SURVEY_ENABLED=
```
Totally irrelevant to this PR but just sneaking this in here.

weerd replied:
😏

mendelB commented:
Sweeet! @lkpttn can we please get your 👨‍⚖️?

weerd commented:
Do we want to formalize fields for this? like `title`, `content`, and `link`? Or is it too soon to commit to those?

mendelB commented:
Yeah, I don't really see why not. It feels like a pretty straightforward set of fields for this action. Shall I go ahead and append that to this PR?

weerd commented:
Ya, PR is pretty small as is, so it'd only be maybe an extra file or two on the PHP side w/ the entity modifications 👍

mendelB commented:
UPDATE:
Added contentful `title`, `content`, and `link` fields for [Share Action](https://app.contentful.com/spaces/81iqaqpfd8fy/content_types/shareAction/fields).

I also updated the few Social Shares we have now, so that when we transition nothing breaks.
Once we deploy this we can remove the `additionalContent` field as well. (I'll add a note to Pivotal)

weerd commented on resources/assets/components/ShareAction/ShareAction.js:
```diff
@@ -21,38 +26,32 @@ const ShareAction = (props) => {
-  }),
+  title: PropTypes.string.isRequired,
+  content: PropTypes.string,
+  link: PropTypes.string.isRequired,
```
🎉
